### PR TITLE
merge version blocks bugfix

### PR DESCRIPTION
**Pull request by guyfreund**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/29822

## Description
See fixing issue for more details.

### CODE REVIEW

**Review by ShahafBenYakir — COMMENTED** [review]
UT

File: Tests/scripts/infrastructure_tests/release_notes_generator_test.py [hunk 1]
```diff
@@ -293,6 +293,24 @@ def test_release_notes_summary_with_ignored_rns(self):
 
 
 class TestMergeVersionBlocks:
+    def test_spaced_content_entity_and_old_format(self):
```
Comment (ShahafBenYakir) [inline Tests/scripts/infrastructure_tests/release_notes_generator_test.py:296]: Given when then

File: Utils/release_notes_generator.py [hunk 2]
```diff
@@ -20,8 +20,9 @@
 
 EMPTY_LINES_REGEX = re.compile(r'\s*-\s*\n')
 IGNORED_LINES_REGEX = re.compile(r'<!-[\W\w]*?-->')
-ENTITY_TYPE_SECTION_REGEX = re.compile(r'^#### (\w+)$\n([\w\W]*?)(?=^#### )|^#### (\w+)$\n([\w\W]*)', re.M)
-ENTITY_SECTION_REGEX = re.compile(r'^##### (.+)$\n([\w\W]*?)(?=^##### )|^##### (.+)$\n([\w\W]*)', re.M)
+ENTITY_TYPE_SECTION_REGEX = re.compile(r'^#### ([\w ]+)$\n([\w\W]*?)(?=^#### )|^#### ([\w ]+)$\n([\w\W]*)', re.M)
+ENTITY_SECTION_REGEX = re.compile(r'^##### (.+)$\n([\w\W]*?)(?=^##### )|^##### (.+)$\n([\w\W]*)|'
+                                  r'^- \*\*(.+)\*\*$\n([\w\W]*)', re.M)
```
Comment (ShahafBenYakir) [inline Utils/release_notes_generator.py:25]: What does this need to catch? `r'^- \*\*(.+)\*\*$\n([\w\W]*)'`

Comment (guyfreund) [reply Utils/release_notes_generator.py:25]: to catch the content entity instance name that is wrapped with `**` like here:
```
#### Incident Fields
- **XDR Alerts**
    - Second
```

File: Tests/scripts/infrastructure_tests/release_notes_generator_test.py [hunk 3]
```diff
@@ -293,6 +293,34 @@ def test_release_notes_summary_with_ignored_rns(self):
...
+        """
+        Given
+        - Two release notes files, one with content entity instance wrapped with **, content entity type contains
+        spaces.
```
Comment (ShahafBenYakir) [inline Tests/scripts/infrastructure_tests/release_notes_generator_test.py:300]: Both are with `**` right?
```suggestion
        - Two release notes files with content entity instance wrapped with ** and entity type contains spaces.
```

File: Tests/scripts/infrastructure_tests/release_notes_generator_test.py [hunk 4]
```diff
@@ -293,6 +293,34 @@ def test_release_notes_summary_with_ignored_rns(self):
...
+        When
+        - We need to merge the two release notes files into one file.
```
Comment (ShahafBenYakir) [inline Tests/scripts/infrastructure_tests/release_notes_generator_test.py:302]:
```suggestion
        - Merging the two release notes files into one file.
```

File: Tests/scripts/infrastructure_tests/release_notes_generator_test.py [hunk 5]
```diff
@@ -293,6 +293,34 @@ def test_release_notes_summary_with_ignored_rns(self):
...
+        Then
+        - Ensure that the content entity instance is wrapped with **.
+        - Ensure that the content entity type contains a whitespace.
```
Comment (ShahafBenYakir) [inline Tests/scripts/infrastructure_tests/release_notes_generator_test.py:305]:
```suggestion
        Then
        - Ensure that the content entity instance is wrapped with **.
        - Ensure that the content entity type contains whitespace.
        - Ensure that the content of both RN files appears in the result file.
```

**Review by ShahafBenYakir — APPROVED** [review]

**Pull request merged**